openstack/compute/v2/servers/requests.go

```diff
@@ -95,6 +96,29 @@ type Network struct {
 	FixedIP string
 }
 
+// Personality is an array of files that are injected into the server at launch.
+type Personality []*File
+
+// File is used within CreateOpts and RebuildOpts to inject a file into the server at launch.
+type File struct {
+	// Path of the file
+	Path string
+	// Contents of the file. Maximum content size is 255 bytes.
+	Contents []byte
+}
+
+// MarshalJSON marshals the escaped file, base64 encoding the contents.
+func (f *File) MarshalJSON() ([]byte, error) {
```

Comment:
Oh, I see; it's implementing the `Marshaler` interface. Can you add a comment above `File` mentioning that? Something like: "File implements the `json.Marshaler` interface, so when a `Create` or `Rebuild` operation is requested, `json.Marshal` will call `File`'s `MarshalJSON method." The only reason for explicitness is because I don't think we have this anywhere else. Also, could you add a `Personality` field to the acceptance test that creates a server?